src/community/script/core/src/main/java/org/geoserver/script/rest/controller/SessionController.java

```diff
@@ -0,0 +1,86 @@
+/** (c) 2014 Open Source Geospatial Foundation - all rights reserved
+ * (c) 2001 - 2013 OpenPlans
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ *
+ * @author David Vick, Boundless 2017
+ **/
+package org.geoserver.script.rest.controller;
+
+import org.geoserver.rest.RestBaseController;
+import org.geoserver.script.rest.service.SessionService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@RestController
+@RequestMapping(path = RestBaseController.ROOT_PATH)
+public class SessionController extends RestBaseController {
+
+    @Autowired
+    SessionService sessionService;
+
+    /**
+     * Get a list of Scripting Sessions
+     * @return
+     */
+    @GetMapping(path = "/sessions", produces = {MediaType.APPLICATION_JSON_VALUE,
+    MediaType.TEXT_PLAIN_VALUE})
+    public void getList(HttpServletResponse response) {
+        sessionService.getSessions(response, null);
+    }
+
+    /**
+     * Get a list of Scripting Sessions
+     * @return
+     */
+    @GetMapping(path = "/sessions/{language}", produces = {MediaType.APPLICATION_JSON_VALUE,
+            MediaType.TEXT_PLAIN_VALUE})
+    public void getList(HttpServletResponse response, @PathVariable String language) {
```

Comment:
Instead of having two methods that do effectively the same thing, you can do the following:

```
    @GetMapping(path = {"/sessions", "/sessions/{language}"}, produces = {MediaType.APPLICATION_JSON_VALUE,
            MediaType.TEXT_PLAIN_VALUE})
    public void getList(HttpServletResponse response, @PathVariable(required = false) String language) {
```

This services both endpoints, if the former is used language will be null.

Reply:
Changed.